src/sonic-yang-models/yang-models/sonic-mirror-session.yang

```diff
@@ -0,0 +1,148 @@
+                key "name";
+
+                leaf name {
+                    type string {
+                        pattern '[a-zA-Z0-9]{1}([-a-zA-Z0-9_]{0,31})';
+                        length 1..32 {
+                            error-message "Invalid mirror session name";
+                            error-app-tag session-name-invalid;
+                        }
+                    }
+                    description
+                        "Mirror session name.";
+                }
+
+                leaf type {
+                    type string {
+                        pattern "ERSPAN|SPAN";
+                    }
+                    default "ERSPAN";
+                    description
+                        "Mirror session type.";
+                }
+
+                leaf src_ip {
+                    type inet:ipv4-address;
+                    description
+                        "ERSPAN source ip. This IP will be set as source ip in
+                        outer header of mirrored frame ";
+                }
+
+                leaf dst_ip {
+                    type inet:ipv4-address;
+                    description
+                        "ERSPAN destination ip. Mirrored frames will be routed to this destination.
+                        This IP will be set as destination ip in outer header of mirrored frame ";
+                }
+
+                leaf gre_type {
+                    type string {
+                        pattern "0[xX][0-9a-fA-F]*";
+                        length 1..6 {
+                            error-message "Invalid GRE type";
+                            error-app-tag gre-type-invalid;
+                        }
+                    }
+                    description
+                        "ERSPAN outer header GRE type.";
+                }
+
+                leaf dscp {
+                    type uint8 {
+                        range "0..63" {
+                            error-message "Invalid dscp value";
+                            error-app-tag dscp-invalid;
+                        }
+                    }
+                    description
+                        "ERSPAN outer header DSCP value. Mirrored frames will be sent with configured DSCP value";
+                }
+
+                leaf ttl {
+                    type uint8 {
+                        range "0..63" {
+                            error-message "Invalid TTL value";
+                            error-app-tag ttl-invalid;
+                        }
+                    }
+                    description
+                        "ERSPAN outer header TTL value. Mirrored frames will be sent with configured TTL value";
+                }
+
+                leaf queue {
+                    type uint8;
+                    description
+                        "ERSPAN Queue. Mirrored frames will be sent to this queue";
+                }
+
+                leaf dst_port {
+                    type leafref {
```

Comment:
Can we add "when" condition to configure this field only when type set to SPAN? the similar condition needs to be done ERSPAN fields as well.

Reply:
Done. added ERSPAN and SPAN conditions. Some fields are valid for both SPAN and ERSPAN.